runtime/onert/backend/train/ops/OperationUtils.cc

```diff
@@ -64,6 +65,32 @@ const IPortableTensor *backpropActivation(const ir::Activation &activation,
   return output_backprop;
 }
 
+void biasGrad(const IPortableTensor *input_backprop, IPortableTensor *bias_grad)
+{
+  assert(bias_grad);
+
+  const nnfw::cker::functor::ReduceOuterDimensions<Eigen::ThreadPoolDevice, float, float, float,
+                                                   Eigen::internal::scalar_sum_op<float>>
+    redux;
+
+  nnfw::cker::Shape input_backprop_shape = getShape(input_backprop);
+  float *input_backprop_buffer = reinterpret_cast<float *>(input_backprop->buffer());
+  const nnfw::cker::Tensor input_backprop_t{input_backprop_shape,
+                                            static_cast<void *>(input_backprop_buffer)};
+
+  nnfw::cker::Shape bias_grad_shape = getShape(bias_grad);
+  float *bias_grad_buffer = getBuffer<float>(bias_grad);
+  nnfw::cker::Tensor bias_grad_t{bias_grad_shape, bias_grad_buffer};
+
+  Eigen::Index outer = input_backprop_shape.Dims(input_backprop_shape.DimensionsCount() - 1);
```

Comment:
These `Eigen` class cannot be hidden?

Until now, onert has hidden these things AFAIK.

```
ONE/runtime/onert/backend (master ✗) grep -rni "eigen"
train/TensorManager.h:38:  // _align is equal to EIGEN_MAX_ALIGN_BYTES, though including Eigen headers
``

Reply:
I hid "Eigen" from onert but now `redux_functor` accepts the following arguments

```cpp
void operator()(const Device &device, const Eigen::Index &outer_dim, const Eigen::Index &inner_dim, const Tensor &input, Tensor *output) const
```

Please let me know if you have some better way to handle this.

Reply:
The built-in type can be transmitted and converted into the `Eigen::Index` in `redux_functor` function.

Reply:
> The built-in type can be transmitted and converted into the `Eigen::Index` in `redux_functor` function.

I implemented @YongseopKim 's suggestion for now :eyes: